src/java.base/share/classes/sun/nio/ch/FileChannelImpl.java

```diff
@@ -1194,10 +1194,7 @@ public long capacity() {
         }
 
         public void unmap() {
-            if (address == 0)
-                return;
             nd.unmap(address, size);
-            address = 0;
```

Comment:
To be safe, would it make sense to keep this code?

For `DirectByteBuffer$Deallocator` you can easily verify that it is safe because the class is only used to create a `Cleaner` (and `Deallocator` is private).

However, here for `Unmapper` `unmap()` is a public method which can be called from other classes through `UnmapperProxy` (currently called by `FileChannelLinesSpliterator#unmap`), and there is also `Unmapper#run` which calls `unmap`. Possibly the current code is still safe and no double-free (?) is possible, but verifying this, and ensuring that this remains the case in the future seems difficult. So this additional safeguard here seems useful (at least in my opinion).

(Sorry for this super late comment)